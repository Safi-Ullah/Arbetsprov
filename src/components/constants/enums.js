export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};