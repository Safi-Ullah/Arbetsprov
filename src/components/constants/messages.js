export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};