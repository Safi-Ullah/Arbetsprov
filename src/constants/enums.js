export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};