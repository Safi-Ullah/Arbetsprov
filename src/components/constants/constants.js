import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};