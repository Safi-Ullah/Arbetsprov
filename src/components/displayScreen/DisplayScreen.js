import React from 'react';
import { Enums } from '../constants';
import { TasksList } from '../tasksList';

export const DisplayScreen = props => {

    const {
        displayConfirmationDialog, displayEditTaskModal,
        selectedScreen, tasks, changeStatus
    } = props;

    let statusToDisplay = Object.keys(Enums.TASK_STATUS);

    if (selectedScreen === Enums.APP_SCREENS.dashboard)
        statusToDisplay = statusToDisplay.filter(status => Enums.TASK_STATUS[status] !== Enums.TASK_STATUS.deleted);
    else
        statusToDisplay = statusToDisplay.filter(status => Enums.TASK_STATUS[status] === Enums.TASK_STATUS.deleted);

    return <div className="container">
        <div className="row main-screen">
            {
                tasks ? statusToDisplay.map((status, index) => <div key={index} className="col-sm">
                    <h6 className="text-center text-capitalize">{ status }</h6>
                    <TasksList
                        changeStatus={changeStatus}
                        displayEditTaskModal={displayEditTaskModal}
                        status={Enums.TASK_STATUS[status]}
                        tasks={tasks.filter(task => task.status === Enums.TASK_STATUS[status])}
                        displayConfirmationDialog={displayConfirmationDialog}/>
                </div>
                ) : null
            }
        </div>
    </div>;
};