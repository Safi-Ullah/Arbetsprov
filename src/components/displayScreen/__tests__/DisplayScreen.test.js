import React from 'react';
import TestRenderer from 'react-test-renderer';
import { DisplayScreen } from '../DisplayScreen';
import { Enums, DEFAULT_STATUS_SORT } from '../../../constants';

const props = {
    displayConfirmationDialog: jest.fn(),
    displayEditTaskModal: jest.fn(),
    selectedScreen: Enums.APP_SCREENS.dashboard,
    tasks: [],
    changeStatus: jest.fn(),
    tasksSortBy: DEFAULT_STATUS_SORT,
    toggleSorting: jest.fn()
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('on dashboard view without any tasks', () => {
        const tree = TestRenderer.create(<DisplayScreen {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('on deleted tasks view without any tasks', () => {
        props.selectedScreen = Enums.APP_SCREENS.deletedTasks;
        const tree = TestRenderer.create(<DisplayScreen {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('on dashboard view with tasks', () => {
        props.tasks = [{
            title: 'test',
            desc: 'test',
            complexity: '0',
            assignee: 'test',
            status: Enums.TASK_STATUS.created
        }];
        const tree = TestRenderer.create(<DisplayScreen {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('on deleted tasks view with tasks', () => {
        props.tasks = [{
            title: 'test',
            desc: 'test',
            complexity: '0',
            assignee: 'test',
            status: Enums.TASK_STATUS.deleted
        }];
        props.selectedScreen = Enums.APP_SCREENS.deletedTasks;
        const tree = TestRenderer.create(<DisplayScreen {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});