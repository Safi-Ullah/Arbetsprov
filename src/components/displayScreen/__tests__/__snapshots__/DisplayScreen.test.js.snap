// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test on dashboard view with tasks 1`] = `
<div
  className="container"
>
  <div
    className="row main-screen"
  >
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          deleted
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          deleted
          .
        </span>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Snapshot test on dashboard view without any tasks 1`] = `
<div
  className="container"
>
  <div
    className="row main-screen"
  >
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          created
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          created
          .
        </span>
      </ul>
    </div>
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          active
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          active
          .
        </span>
      </ul>
    </div>
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          completed
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          completed
          .
        </span>
      </ul>
    </div>
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          onhold
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          onhold
          .
        </span>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Snapshot test on deleted tasks view with tasks 1`] = `
<div
  className="container"
>
  <div
    className="row main-screen"
  >
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          deleted
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <li
          className="list-group-item"
        >
          <button
            className="btn btn-link text-break"
            onClick={[Function]}
          >
            test
          </button>
          <span
            className="badge badge-info"
          >
            0
          </span>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            className="close ml-2"
            data-toggle="dropdown"
            id="dropdownMenuButton"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ⁝
            </span>
          </button>
          <div
            aria-labelledby="dropdownMenuButton"
            className="dropdown-menu"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
            >
              Edit
            </button>
            <div
              className="dropdown-divider"
            />
            <span
              className="dropdown-subheading"
            >
              Change Status
            </span>
            <button
              className="dropdown-item text-capitalize"
              disabled={false}
              onClick={[Function]}
            >
              created
            </button>
            <button
              className="dropdown-item text-capitalize"
              disabled={false}
              onClick={[Function]}
            >
              onhold
            </button>
          </div>
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="delete-btn"
            >
              ×
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Snapshot test on deleted tasks view without any tasks 1`] = `
<div
  className="container"
>
  <div
    className="row main-screen"
  >
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <h6
          className="text-center mt-2 text-capitalize col-sm-11"
        >
          deleted
        </h6>
        <span
          className="d-inline-block"
          tabIndex="0"
          title="Sort by complexity"
        >
          <button
            className="close col-sm-1"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ▾
            </span>
          </button>
        </span>
      </div>
      <ul
        className="list-group"
      >
        <span
          className="text-center text-muted"
        >
          No tasks 
          deleted
          .
        </span>
      </ul>
    </div>
  </div>
</div>
`;
